torch/lib/TH/generic/THTensorMath.c

```diff
@@ -3203,10 +3200,7 @@ void THTensor_(logspace)(THTensor *r_, real a, real b, long n)
   }
 
   if(n == 1) {
-    TH_TENSOR_APPLY(real, r_,
-        *r__data = TH_MATH_NAME(pow)(10.0, a);
-        i++;
-        );
+    THTensor_(set1d)(r_, 0, TH_MATH_NAME(pow)(10.0, a));
```

Comment:
Note that `set1d` does in-bound checks that are unnecessary

Reply:
ya but this is literally for n==1 in linspace and logspace. i think it's fine.